app/models/spree/subscription.rb

```diff
@@ -226,11 +226,13 @@ def add_shipping_costs_to_order(order)
       end
 
       def add_payment_method_to_order(order)
-        if order.payments.exists?
-          order.payments.first.update(source: source, payment_method: source.payment_method)
-        else
-          order.payments.create(source: source, payment_method: source.payment_method, amount: order.total)
-        end
+        credit_card = attach_credit_card
+        payment_method = attach_payment_method
+      
+        payment_method.stores = Spree::Store.all
```

Comment:
payment method data preparation is not dynamic so this should be in seeds or rake task, it doesn't make sense here.